recipes/yarr/meta.yaml

```diff
@@ -0,0 +1,63 @@
+{% set name = "yarr" %}
+{% set version = "1.5.2" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://gitlab.cern.ch/yarr/{{ name }}/-/archive/v{{ version }}/{{ name }}-v{{ version }}.tar.gz
+  sha256: 56dd0a42f5260470239f76b89aad016913e8267fbf3e10227abf1797618de18d
+  patches:
+    - patches/0001-yarr_873.patch
+    - patches/0002-patch-strip-debug.patch
+    - patches/0003-patch-dependency-resolution-for-osx.patch
+
+build:
+  skip: true  # [win]
+  script:
+    - CXXFLAGS="${CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY" cmake ${CMAKE_ARGS} -DCMAKE_INSTALL_PREFIX=$PREFIX -S "${SRC_DIR}" -B build  # [osx]
+    - CXXFLAGS="${CXXFLAGS}" cmake ${CMAKE_ARGS} -DCMAKE_INSTALL_PREFIX=$PREFIX -S "${SRC_DIR}" -B build  # [linux]
+    - cmake --build build --clean-first --parallel="${CPU_COUNT}"
+    - cmake --install build
+    - mkdir -p $PREFIX/share/yarr
+    - rsync -ravz $SRC_DIR/configs $PREFIX/share/yarr
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ stdlib('c') }}
+    - cmake
+    - make
+    - rsync
+  host: []
+  run: []
+
+test:
+  commands:
+    - test -f $PREFIX/bin/scanConsole
+    - test -f $PREFIX/share/yarr/configs/controller/emuCfg_rd53a.json
+    - scanConsole -h
+    - scanConsole -k
+    # needed for YARR to pick up the configs when it loads the emulator
+    - ln -s "$PREFIX/share/yarr/configs" ./configs  # [linux]
```

Comment:
Is this expected? Or should the sources be patched to be more friendly to packaging?

Reply:
this is generally what users would need to do on some level when they pull this. the `emulator` requires looking for a `configs` path specifically with the information needed 

```
            "feCfg" : "configs/emulator/emu_fe0_rd53a.json"
```

and it's not trivial to patch the source here as this touches a few more places (and not really worthwhile, I don't think).

Reply:
You could do something lik:

```
find $PREFIX/share/yarr/configs -name '*.json' -print -exec sed -i 's@"configs/@"'$PREFIX'/configs/@g' {} \;
```

Conda will then take care of replacing the path at install time to be correct.

Reply:
added it, that's a great idea and I did not know conda could do that :)

Reply:
@chrisburr seems like while the regex works, `conda` is stripping away the full prefix on install? I'm not fully sure why though...

Reply:
I can confirm that the regex works to replace as

```
{
    "ctrlCfg" : {
        "type": "emu_Rd53a",
        "cfg" : {
            "type" : "rd53a",
            "tx" : {
                "id" : 1337,
                "size" : 256
            },
            "rx" : {
                "id" : 1338,
                "size" : 256
            },
            "feCfg" : "$PREFIX/configs/emulator/emu_fe0_rd53a.json"
        }
    }
}
```

but looking at conda's docs: https://docs.conda.io/projects/conda-build/en/stable/resources/package-spec.html#package-metadata

it seems I need to be using the actual value of the prefix, not the `"$PREFIX"`, but the value of it. It also seems to indicate as such here: https://github.com/conda/conda-prefix-replacement

Reply:
will try this version

```
find $PREFIX/share/yarr/configs -name '*.json' -print -exec sed -i "s@\"configs/@\"$PREFIX/configs/@g" {} +
```

for the replacement and see if that works.

Reply:
I think the problem is a missing `share/yarr/` for the prefix substitution.. so let me try that first.